pkg/cleaner/finalizer_cleaner.go

```diff
@@ -0,0 +1,339 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cleaner
+
+import (
+	"fmt"
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+	"time"
+)
+
+const (
+	finalizerCheckPerdiodTime = 1 * time.Second
+	finalizerCheckTimeout     = 30 * time.Second
+)
+
+type FinalizerCleaner struct {
+	client sc.Interface
+}
+
+// NewFinalizerCleaner returns new pointer to FinalizerCleaner
+func NewFinalizerCleaner(scClient sc.Interface) *FinalizerCleaner {
+	return &FinalizerCleaner{scClient}
+}
+
+// RemoveFinalizers removes specific finalizers from all ServiceCatalog CRs
+func (fc *FinalizerCleaner) RemoveFinalizers() error {
+	klog.V(4).Info("Removing finalizers from ClusterServiceBrokers")
+	err := removeFinalizerFromClusterServiceBroker(fc.client)
+	if err != nil {
+		return fmt.Errorf("failed during removing ClusterServiceBroker finalizer: %s", err)
```

Comment:
in Service Catalog there is a `pretty` maybe you can use it here for each statement and replace e.g. `ClusterServiceBroker` with `pretty.ClusterServiceBroker`